src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -453,7 +453,17 @@ private async Task<bool> ReadFromQueue(QueueId myQueueId, IQueueAdapterReceiver
                 if (pubSubCache.TryGetValue(streamId, out streamData))
                 {
                     streamData.RefreshActivity(now);
-                    StartInactiveCursors(streamData, startToken); // if this is an existing stream, start any inactive cursors
+                    if (streamData.StreamRegistered)
+                    {
+                        StartInactiveCursors(streamData,
+                            startToken); // if this is an existing stream, start any inactive cursors
+                    }
+                    else
+                    {
+                        this.logger.LogWarning($"Pulled new messages in stream {streamId} from the queue, but pulling agent haven't succeeded in" +
+                                                   $"RegisterStream yet, will start deliver on this stream after RegisterStream succeeded");
```

Comment:
This warning could spam a lot..
Maybe instead add a check in CleanupPubSubCache to log any stream that is still unregistered?
Other options?

Reply:
hmm maybe we log it as Debug? @sergeybykov what's your opinion? 

> Maybe instead add a check in CleanupPubSubCache to log any stream that is still unregistered?

not sure why this is related. Can you explain more?

Reply:
Yeah, probably doesn't belong in CleanupPubSubCache.  The reason I suggested that is because that check is done less often, so we'd not span the logs.. but conceptually, it makes no sense :/